Boost: Update getting started and upgrade pages copy

dilirity opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Fixes #36472

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Update getting started features list;
* Update upgrade copy.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

pbNhbs-aae-p2

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

no

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Check `/wp-admin/admin.php?page=jetpack-boost#getting-started` and `/wp-admin/admin.php?page=jetpack-boost#upgrade` to make sure they match the proposed design changes.

dilirity committed:
Update getting started features list

dilirity committed:
Update upgrade screen copy

dilirity committed:
add changelog

dilirity committed:
Update label

thingalon reviewed (changes requested):
Thanks for tackling this @dilirity! Looking good.

A few details that don't match @augurlu's copy for the getting started page:

- The free "Manual" option for Auto CSS Generation is supposed to have a ❌ 
- Image CDN Quality settings is meant to be a separate entry of its own, below Dedicated email support

dilirity committed:
Fix showing "auto css optimization" as included in the free version

dilirity commented:
Thanks @thingalon!

I've fixed the Manual part.

For the `Image CDN Quality settings` - it is a separate entry, but I just added `Page Cache` to the list (right above it), since it was missing.

thingalon reviewed (approved):
Nice! Thanks @dilirity  -- 🚢
